CRM/Event/Form/Participant.php

```diff
@@ -1228,9 +1228,7 @@ public function submit($params) {
 
     if ($this->_mode) {
       // add all the additional payment params we need
-      $this->_params["state_province-{$this->_bltID}"] = $this->_params["billing_state_province-{$this->_bltID}"] = CRM_Core_PseudoConstant::stateProvinceAbbreviation($this->_params["billing_state_province_id-{$this->_bltID}"]);
-      $this->_params["country-{$this->_bltID}"] = $this->_params["billing_country-{$this->_bltID}"] = CRM_Core_PseudoConstant::countryIsoCode($this->_params["billing_country_id-{$this->_bltID}"]);
-
+      $this->_params = $this->prepareParamsForPaymentProcessor($this->_params);
```

Comment:
Does it matter that now these are only being set if `!empty($params["billing_state_province_id-{$this->_bltID}"])` and similarly for country id?
 
and that
$params['state_province']
and 
 $params['country']

are now being set as well? I assume not but just wanted to double check.

Reply:
Or is the fact that we were not previous testing for empty values the problem!

Reply:
@stesi561 The fact we weren't checking for empty values would potentially cause PHP notices.  And missing params that are now being set means we might now be saving data that might not have been saved previously (or the payment processor had to implement custom handling to look for the parameter) - eg. for Country.

Reply:
I assume this change won't clash with custom handling in payment processors. The dummy processor was already saving address data. However if this is a change of behaviour I think it's possibly worth referencing in the PR description?